be/src/runtime/runtime_filter_worker.cpp

```diff
@@ -61,17 +63,24 @@ void RuntimeFilterPort::publish_runtime_filters(std::list<vectorized::RuntimeFil
 
     for (auto* rf_desc : rf_descs) {
         auto* filter = rf_desc->runtime_filter();
-        if (filter == nullptr || !rf_desc->has_remote_targets() || rf_desc->merge_nodes().size() == 0) continue;
 
-        // for broadcast join, we need to send only one copy.
-        // and we have already assigned sender frag inst id in planner.
-        if (rf_desc->join_mode() == TRuntimeFilterBuildJoinMode::BORADCAST) {
-            if (rf_desc->sender_finst_id() != state->fragment_instance_id()) {
-                continue;
-            }
-            VLOG_FILE << "RuntimeFilterPort::publish_runtime_filters. broadcast join filter_id = "
-                      << rf_desc->filter_id() << ", finst_id = " << rf_desc->sender_finst_id();
-        }
+        if (filter == nullptr || !rf_desc->has_remote_targets()) continue;
+
+        auto directly_send_broadcast_grf = rf_desc->join_mode() == TRuntimeFilterBuildJoinMode::BORADCAST &&
+                                           !rf_desc->broadcast_grf_senders().empty();
+        // when send grf generated by broadcast join, GRF coordinator is needless.
+        if (!directly_send_broadcast_grf && rf_desc->merge_nodes().empty()) continue;
+
+        // for non-broadcast join, each fragment instance must send grf
+        // for broadcast join, if directly sending(not via GRF coordinator) is adopted, multiple fragment instances
+        // are chosen to send grf copies, otherwise, send only one copy. the senders are planned by FE.
+        auto need_sender_grf = rf_desc->join_mode() != TRuntimeFilterBuildJoinMode::BORADCAST ||
+                               rf_desc->broadcast_grf_senders().count(state->fragment_instance_id()) ||
+                               rf_desc->sender_finst_id() == state->fragment_instance_id();
+        if (!need_sender_grf) continue;
+
+        VLOG_FILE << "RuntimeFilterPort::publish_runtime_filters. broadcast join filter_id = " << rf_desc->filter_id()
```

Comment:
```suggestion
        VLOG_FILE << "RuntimeFilterPort::publish_runtime_filters. join filter_id = " << rf_desc->filter_id()
```
When arriving at this code, maybe it isn't a broadcast join.

Reply:
ok